src/Core/src/Response.php

```diff
@@ -157,6 +268,33 @@ public function getStatusCode(): int
 
     public function toStream(): ResultStream
     {
+        if ($this->streamStarted) {
+            throw new RuntimeException('Can not create a ResultStream because the body started being downloaded. Do not call Result::multiplex');
+        }
+
```

Comment:
We should add `$this->bodyDownloaded = true;`, right? 

We don't know for sure. But we have started streaming the response... or how do we handle this?

Reply:
we don't know if the body is downloaded...

But you're right, better saying that we're done on our side